barretenberg/cpp/pil/vm2/alu.pil

```diff
@@ -201,27 +204,101 @@ sel_mul_u128 * (1 - sel_tag_err)
     ) = 0;
 
 // TODO: Once lookups support expression in tuple, we can inline constant_64 into the lookup.
-// Note: only used for MUL, so gated by sel_op_mul
+// Note: only used for MUL/DIV u128, so gated by sel_mul_div_u128
 pol commit constant_64;
-sel_op_mul * (64 - constant_64) = 0;
+sel_mul_div_u128 * (64 - constant_64) = 0;
 
 #[RANGE_CHECK_MUL_U128_A_LO]
-sel_mul_u128 { a_lo, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
+sel_mul_div_u128 { a_lo, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
 
 #[RANGE_CHECK_MUL_U128_A_HI]
-sel_mul_u128 { a_hi, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
+sel_mul_div_u128 { a_hi, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
 
 #[RANGE_CHECK_MUL_U128_B_LO]
-sel_mul_u128 { b_lo, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
+sel_mul_div_u128 { b_lo, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
 
 #[RANGE_CHECK_MUL_U128_B_HI]
-sel_mul_u128 { b_hi, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
+sel_mul_div_u128 { b_hi, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
 
 // No need to range_check c_hi for cases other than u128 because we know a and b's size from the tags and have looked
 // up max_value. i.e. we cannot provide a malicious c, c_hi such that a + b - c_hi * 2^n = c passes for n < 128.
 // No need to range_check c_lo = ic because the memory write will ensure ic <= max_value.
 #[RANGE_CHECK_MUL_U128_C_HI]
-sel_mul_u128 { c_hi, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
+sel_mul_div_u128 { c_hi, constant_64 } in range_check.sel { range_check.value, range_check.rng_chk_bits };
+
+// DIV
+
+sel_op_div * (1 - sel_op_div) = 0;
+
+// TODO
+// if sel_op_div = 1 && ib = 0, div_0_error
+
+// This is integer division, so we disallow FF (use tag err here?)
+sel_op_div * sel_is_ff = 0;
```

Comment:
Not sure whether `FF` inputs for `DIV` should fail a relation or be counted as a `TAG_ERROR`? If the latter, I'll check against `sel_tag_err` and add a 'case' alongside `AB_TAGS_EQ`.